src/ray/rpc/grpc_server.h

```diff
@@ -100,12 +112,30 @@ class GrpcServer {
   /// `GrpcServer`, as it holds the underlying `grpc::Service`.
   ///
   /// \param[in] service A `GrpcService` to register to this server.
-  void RegisterService(GrpcService &service);
+  /// NOTE: if token_auth is not set to false, cluster_id_ must not be Nil.
+  void RegisterService(GrpcService &service, bool token_auth = true);
   void RegisterService(grpc::Service &service);
 
   grpc::Server &GetServer() { return *server_; }
 
+  ClusterID const GetClusterId() {
```

Comment:
```suggestion
  const ClusterID GetClusterId() const {
```

Reply:
Had to make the method non-const because acquiring a mutex is non-const